src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,473 @@
+tests pass. 
+You don't need to implement `main()`.
+
+To simulate asynchronous operations, your code will call the following functions, which are 
+provided for you:
+
+<!-- dart annotation format borrowed from  
+https://api.dartlang.org/stable/2.4.0/dart-developer/dart-developer-library.html -->
+<!-- See e.g. getCurrentTag() → UserTag -->
+
+{% prettify dart %}
+getRole() → Future<String> 
+// Returns a short description of the user's role
+
+getLoginAmount() → Future<int> 
+// Returns the number of times a user has logged in
+{% endprettify %}
+#### Part 1: `reportUserRole()`
+
+Add logic to the `reportUserRole()` function:
+<!-- Some bulleted items are intentionally lacking punctuation to avoid confusing the users about characters in string values -->
+* `reportUserRole()` returns a future that completes with the String: `"User role: <user role>"`
+  * Example return value from `reportUserRole()`: `"User role: tester"` 
+  * Note: you must use the actual value returned by `getRole()`; copying and pasting the example return value won't make the test pass. 
+* Get the user role by calling the provided function `getRole()` 
+
+####  Part 2: `reportLogins()`
+Implement an `async` function `reportLogins()`:
+* `reportLogins()` returns the string `"Total number of logins: <# of logins>"`.
+  * Example return value from `reportLogins()`: `"Total number of logins: 57"` 
+  * Note: you must use the actual value returned by `getLoginAmount()`; copying and pasting the example return value won't make the test pass.
+* Get the number of logins by calling the provided function `getLoginAmount()`
+
+<!-- [//]: https://gist.github.com/f751b692502c4ee43d932f745860b056 -->
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new-dart.html?id=f751b692502c4ee43d932f745860b056&theme=dark"
+  frameborder="no"
+  height="550"
+  width="100%">
+</iframe>
+
+  {{ site.alert.info }}
+    If your tests pass, you can ignore [Info-level messages.](/guides/language/analysis-options#customizing-analysis-rules)
+  {{ site.alert.end }}
+
+## Handling errors
+Handle errors in an `async` function by using try-catch: 
+
+```dart
+  try {
+    var order = await getUserOrder();
+    print('Awaiting user order...');
+  } catch (err) {
+    print('Caught error: $err');
+  }
+```
+Within an `async` function, you can write [try-catch clauses](/guides/language/language-tour#catch)
+the same way for asynchronous and synchronous code. 
+<!-- To handle the error, use the await keyword within a try block -->
+<!-- Exception vs. Error -->
+
+### Example: async and await with try-catch
+Run the following example to see how to handle an error from an 
+asynchronous function. Can you guess what the output will be before running it?
+<!-- [//]: https://gist.github.com/25ade03f0632878a9169209e3cd7bef2 -->
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new-dart.html?id=25ade03f0632878a9169209e3cd7bef2"
```

Comment:
The code says "Imagine that this function... throws an exception." But it does throw an exception!!! Change to "... more complex." ?